app/services/proofing/resolution/plugins/threat_metrix_plugin.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Proofing
+  module Resolution
+    module Plugins
+      class ThreatMetrixPlugin
+        def call(
+          applicant_pii:,
+          current_sp:,
+          request_ip:,
+          threatmetrix_session_id:,
+          timer:,
+          user_email:
+        )
+          unless FeatureManagement.proofing_device_profiling_collecting_enabled?
+            return threatmetrix_disabled_result
+          end
+
+          # The API call will fail without a session ID, so do not attempt to make
+          # it to avoid leaking data when not required.
+          return threatmetrix_id_missing_result if threatmetrix_session_id.blank?
+          return threatmetrix_pii_missing_result if applicant_pii.blank?
+
+          ddp_pii = applicant_pii.dup
+          ddp_pii[:threatmetrix_session_id] = threatmetrix_session_id
+          ddp_pii[:email] = user_email
+          ddp_pii[:request_ip] = request_ip
+
+          timer.time('threatmetrix') do
+            proofer.proof(ddp_pii)
+          end.tap do |result|
+            Db::SpCost::AddSpCost.call(
+              current_sp, :threatmetrix,
+              transaction_id: result.transaction_id
+            )
```

Comment:
is the idea each plugin will log its own cost? unsure if there are plans to consolidate in the future but maybe there's a world where cost information is returned as part of the result of each plugin and then it's iterated over at the end (so like, if a new plugin is added, but doesn't add cost attributes to its result, that would fail in specs because the proofer is trying to read methods that aren't there or something)

Reply:
Yeah I would say there are lots of options for how plugins interact and how we handle this. Right now just focused on the lift 'n ship, as you said